modules/video/src/bgfg_gaussmix2.cpp

```diff
@@ -905,24 +898,47 @@ void BackgroundSubtractorMOG2Impl::getBackgroundImage(OutputArray backgroundImag
                     break;
             }
             float invWeight = 1.f/totalWeight;
-            switch(nchannels)
-            {
-            case 1:
-                meanBackground.at<uchar>(row, col) = (uchar)(meanVal[0] * invWeight);
-                meanVal[0] = 0.f;
-                break;
-            case 3:
-                Vec3f& meanVec = *reinterpret_cast<Vec3f*>(&meanVal[0]);
-                meanBackground.at<Vec3b>(row, col) = Vec3b(meanVec * invWeight);
-                meanVec = 0.f;
-                break;
-            }
+
+            meanBackground.at<Vec<T,CN> >(row, col) = Vec<T,CN>(meanVal * invWeight);
+            meanVal = 0.f;
+
             firstGaussianIdx += nmixtures;
         }
     }
     meanBackground.copyTo(backgroundImage);
 }
 
+void BackgroundSubtractorMOG2Impl::getBackgroundImage(OutputArray backgroundImage) const
+{
+    CV_Assert(frameType == CV_8UC1 || frameType == CV_8UC3 || frameType == CV_32FC1 || frameType == CV_32FC3);
+
+#ifdef HAVE_OPENCL
+    if (opencl_ON)
+    {
+        CV_OCL_RUN(opencl_ON, ocl_getBackgroundImage(backgroundImage))
+
+        opencl_ON = false;
+        return;
```

Comment:
Actually I didn't add this code, getBackgroundImage and the return where already there. I just added getBackgroundImage_intern before the method, and moved some of the content from getBackgroundImage to getBackgroundImage_intern, therefore it looks like getBackgroundImage is new.
I just looked at the definition of CV_OCL_RUN I saw that it does make a return when required, so indeed we don't need the return here. I remove it.